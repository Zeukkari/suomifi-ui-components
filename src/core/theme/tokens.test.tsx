import React from 'react';
import { render } from 'react-testing-library';
import { axeTest } from '../../utils/test/axe';

import { defaultTokens } from './';
import { Expander } from '../../';

const { colors } = defaultTokens;
const customColors = {
  ...colors,
  highlightBase: 'hsl(1, 2%, 3%)',
};

const Test = (
  <Expander.group
    OpenAll={'Open all'}
    CloseAll={'Close all'}
    tokens={{ colors: customColors }}
  >
    <Expander title="Test expander 1">Test expander content 1</Expander>
    <Expander title="Test expander 2">Test expander content 2</Expander>
    <Expander title="Test expander 3">Test expander content 3</Expander>
  </Expander.group>
);

test('snapshot testing', () => {
  const TestRendered = render(Test);
  const { container } = TestRendered;
  expect(container.firstChild).toMatchSnapshot();
});

test('should not have basic accessibility issues', axeTest(Test));
