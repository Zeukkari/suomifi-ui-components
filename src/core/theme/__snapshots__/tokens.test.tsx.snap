// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
.c5 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c5:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c5::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c5::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c5::-webkit-inner-spin-button {
  height: auto;
}

.c5::-webkit-outer-spin-button {
  height: auto;
}

.c2 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: block;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c9 {
  display: inline-block;
  vertical-align: baseline;
}

.c4.fi-button--disabled {
  cursor: not-allowed;
}

.c1 {
  display: block;
  width: 100%;
  max-width: 100%;
}

.c7 {
  display: block;
  width: 100%;
  max-width: 100%;
}

.c10 {
  display: none;
}

.c8,
.c8 * {
  cursor: pointer;
}

.c8:hover,
.c8:active,
.c8:focus,
.c8:focus-within {
  cursor: pointer;
}

.c3 {
  color: hsl(0,0%,16%);
  border-radius: 2px;
  cursor: pointer;
}

.c3:focus {
  outline: 0;
  position: relative;
}

.c3:focus:after {
  content: '';
  position: absolute;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
  border-radius: 4px;
  background-color: transparent;
  border: 1px solid hsl(23,82%,53%);
  box-sizing: border-box;
  box-shadow: 0 0 10px 0 hsl(23,82%,53%);
  z-index: 9999;
}

.c3:focus:not(:focus-visible):after {
  content: none;
}

.c0 {
  color: hsl(0,0%,16%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 > .fi-expander-group_panels {
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
}

.c0 > .fi-expander-group_panels .fi-expander {
  margin-top: 0;
  margin-bottom: 0;
  -webkit-transition: margin 50ms cubic-bezier(0.28,0.84,0.42,1);
  transition: margin 50ms cubic-bezier(0.28,0.84,0.42,1);
}

.c0 > .fi-expander-group_panels .fi-expander.fi-expander--open:not(:first-of-type) {
  margin-top: 16px;
}

.c0 > .fi-expander-group_panels .fi-expander.fi-expander--open:not(:last-of-type) {
  margin-bottom: 16px;
}

.c0 > .fi-expander-group_all-button {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
  margin-left: auto;
  margin-bottom: 8px;
  padding: 4px 0;
  color: hsl(1,2%,3%);
  cursor: pointer;
}

.c0 > .fi-expander-group_all-button:focus {
  outline: 0;
  position: relative;
}

.c0 > .fi-expander-group_all-button:focus:after {
  content: '';
  position: absolute;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
  border-radius: 4px;
  background-color: transparent;
  border: 1px solid hsl(23,82%,53%);
  box-sizing: border-box;
  box-shadow: 0 0 10px 0 hsl(23,82%,53%);
  z-index: 9999;
}

.c0 > .fi-expander-group_all-button:focus:not(:focus-visible):after {
  content: none;
}

.c6 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  background-color: hsl(0,0%,100%);
  position: relative;
  position: relative;
  padding: 0;
  border-radius: 2px;
  box-shadow: 0 1px 2px 0 rgba(41,41,41,0.14),0 1px 5px 0 rgba(41,41,41,0.12);
}

.c6:before {
  content: '';
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  width: 100%;
  height: 100%;
}

.c6:before {
  background-color: hsl(212,63%,98%);
  opacity: 0;
}

.c6.fi-expander--open:before {
  opacity: 1;
  -webkit-transition: opacity 50ms cubic-bezier(0.28,0.84,0.42,1);
  transition: opacity 50ms cubic-bezier(0.28,0.84,0.42,1);
}

.c6 .fi-expander_title {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
  line-height: 1;
  position: relative;
  display: block;
  width: 100%;
}

.c6 .fi-expander_title:focus {
  outline: 0;
  position: relative;
}

.c6 .fi-expander_title:focus:after {
  content: '';
  position: absolute;
  top: -4px;
  right: -4px;
  bottom: -4px;
  left: -4px;
  border-radius: 4px;
  background-color: transparent;
  border: 1px solid hsl(23,82%,53%);
  box-sizing: border-box;
  box-shadow: 0 0 10px 0 hsl(23,82%,53%);
  z-index: 9999;
}

.c6 .fi-expander_title:focus:not(:focus-visible):after {
  content: none;
}

.c6 .fi-expander_title--no-tag {
  padding-top: 16px;
  padding-right: 64px;
  padding-bottom: 16px;
  padding-left: 16px;
  color: hsl(212,63%,45%);
}

.c6 .fi-expander_title-icon {
  position: absolute;
  top: 0;
  right: 0;
  margin: 16px;
}

.c6 .fi-expander_title--open .fi-expander_title-icon,
.c6 .fi-expander_title-icon--open {
  -webkit-transform: rotate(-180deg);
  -ms-transform: rotate(-180deg);
  transform: rotate(-180deg);
}

.c6 > .fi-expander_content {
  position: relative;
  display: block;
  height: 0;
  overflow: hidden;
  -webkit-transform: scaleY(0);
  -ms-transform: scaleY(0);
  transform: scaleY(0);
  -webkit-transform-origin: top;
  -ms-transform-origin: top;
  transform-origin: top;
  -webkit-transition: all 50ms cubic-bezier(0.28,0.84,0.42,1);
  transition: all 50ms cubic-bezier(0.28,0.84,0.42,1);
  will-change: transition,height;
}

.c6 > .fi-expander_content:not(.fi-expander_content--no-padding) {
  padding: 0 16px;
}

.c6 > .fi-expander_content.fi-expander_content--open {
  height: 10%;
  overflow: visible;
  -webkit-animation: fi-expander_content-anim 50ms cubic-bezier(0.28,0.84,0.42,1) 1 forwards;
  animation: fi-expander_content-anim 50ms cubic-bezier(0.28,0.84,0.42,1) 1 forwards;
}

.c6 > .fi-expander_content.fi-expander_content--open:not(.fi-expander_content--no-padding) {
  padding-top: 0;
  padding-right: 16px;
  padding-bottom: 16px;
  padding-left: 16px;
}

<div
  class="c0 fi-expander-group c1 c2"
>
  <button
    aria-disabled="false"
    class="fi-button fi-expander-group_all-button c3 fi-button c4 c5"
    tabindex="0"
    tokens="[object Object]"
    type="button"
  >
    Open all
  </button>
  <div
    class="fi-expander-group_panels c2"
  >
    <div
      class="c6 fi-expander c7 c2"
    >
      <button
        aria-disabled="false"
        aria-expanded="false"
        class="fi-button fi-expander_title fi-expander_title--no-tag c8 c4 c5"
        tabindex="0"
        type="button"
      >
        Test expander 1
        <svg
          aria-hidden="true"
          class="fi-expander_title-icon c9"
          fill="hsl(212, 63%, 45%)"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M13 15.6c-.3.3-.6.4-1 .4s-.7-.1-1-.4l-5.6-5.3c-.5-.5-.5-1.4 0-1.9s1.4-.5 2 0l4.6 4.4 4.6-4.4c.5-.5 1.4-.5 2 0 .5.5.5 1.4 0 1.9L13 15.6z"
          />
        </svg>
      </button>
      <div
        aria-hidden="true"
        class="fi-expander_content c10 fi-expander c7 c2"
        hidden=""
      >
        Test expander content 1
      </div>
    </div>
    <div
      class="c6 fi-expander c7 c2"
    >
      <button
        aria-disabled="false"
        aria-expanded="false"
        class="fi-button fi-expander_title fi-expander_title--no-tag c8 c4 c5"
        tabindex="0"
        type="button"
      >
        Test expander 2
        <svg
          aria-hidden="true"
          class="fi-expander_title-icon c9"
          fill="hsl(212, 63%, 45%)"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M13 15.6c-.3.3-.6.4-1 .4s-.7-.1-1-.4l-5.6-5.3c-.5-.5-.5-1.4 0-1.9s1.4-.5 2 0l4.6 4.4 4.6-4.4c.5-.5 1.4-.5 2 0 .5.5.5 1.4 0 1.9L13 15.6z"
          />
        </svg>
      </button>
      <div
        aria-hidden="true"
        class="fi-expander_content c10 fi-expander c7 c2"
        hidden=""
      >
        Test expander content 2
      </div>
    </div>
    <div
      class="c6 fi-expander c7 c2"
    >
      <button
        aria-disabled="false"
        aria-expanded="false"
        class="fi-button fi-expander_title fi-expander_title--no-tag c8 c4 c5"
        tabindex="0"
        type="button"
      >
        Test expander 3
        <svg
          aria-hidden="true"
          class="fi-expander_title-icon c9"
          fill="hsl(212, 63%, 45%)"
          focusable="false"
          height="1em"
          viewBox="0 0 24 24"
          width="1em"
        >
          <path
            d="M13 15.6c-.3.3-.6.4-1 .4s-.7-.1-1-.4l-5.6-5.3c-.5-.5-.5-1.4 0-1.9s1.4-.5 2 0l4.6 4.4 4.6-4.4c.5-.5 1.4-.5 2 0 .5.5.5 1.4 0 1.9L13 15.6z"
          />
        </svg>
      </button>
      <div
        aria-hidden="true"
        class="fi-expander_content c10 fi-expander c7 c2"
        hidden=""
      >
        Test expander content 3
      </div>
    </div>
  </div>
</div>
`;
